lib/matplotlib/artist.py

```diff
@@ -385,15 +385,42 @@ def get_children(self):
         """
         return []
 
+    def _default_contains(self, mouseevent, canvas=None):
+        """Base impl. for checking whether a mouseevent happened in an artist.
+
+        1. If the artist defines a custom checker, use it.
+        2. If the event did not occur in the same canvas, reject it.
+        3. Otherwise, return `None, {}`, indicating that the subclass'
+           implementation should be used.
+
+        Subclasses should start their definition of `contains` as follows:
+
+            inside, info = self._default_contains(mouseevent)
+            if inside is not None:
+                return inside, info
+            # subclass-specific implementation follows
+
+        The `canvas` kwarg is provided for the implementation of
+        `Figure.contains`.
+        """
+        if callable(self._contains):
+            return self._contains(self, mouseevent)
+        if canvas is None:
+            canvas = self.figure.canvas
```

Comment:
I could pass `figure` as a kwarg instead of `canvas but it comes down to the same, right?  It's just so that `figure.contains(event)` does the right thing.